aptos-move/framework/aptos-stdlib/sources/simple_map.move

```diff
@@ -77,6 +77,25 @@ module aptos_std::simple_map {
         vector::push_back(&mut map.data, Element { key, value });
     }
 
+    /// Insert key/value pair or update an existing key to a new value
+    public fun upsert<Key: store + drop + copy, Value: store + drop + copy>(
```

Comment:
I wonder if it's valuable to provide the inline upsert version for when Key/Value are not droppable. That might still be useful IMO. What do you think @lightmark ?

Reply:
imo, value does not have to be dropable, but key should be. Two kv pairs sharing the same key w/o copy + drop makes little sense to me. 
This function can return an `Option<Value>`. If `Value: drop` we can just ignore the returned value.

Reply:
The problem is that without the native "swap" i proposed we can't still do this. 

element.value = *value

requires copy and drop. We cannot express swap(&mut element.value, value), it needs native support.

So this would require push_back, vector::swap flow, in which case we can make the types the same for add, which is nice for consistency. So I go for that and return (option<Key>, option<Value>) which is superior to my previous lambda